cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/engine/sql/SQLEngine.java

```diff
@@ -16,14 +16,11 @@
 
 package io.cdap.cdap.etl.api.engine.sql;
 
+import com.google.common.base.Strings;
```

Comment:
avoid putting guava classes in api classes to avoid conflicts. 

Also, does this mean guava is a dependency of cdap-etl-api? I thought guava had been removed from all api modules.

Reply:
It's weird, this is the dependency tree I see:

```
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ cdap-etl-api ---
[INFO] io.cdap.cdap:cdap-etl-api:jar:6.5.0-SNAPSHOT
[INFO] +- io.cdap.cdap:cdap-api:jar:6.5.0-SNAPSHOT:compile
[INFO] |  +- io.cdap.cdap:cdap-api-common:jar:6.5.0-SNAPSHOT:compile
[INFO] |  +- com.google.code.findbugs:jsr305:jar:2.0.1:compile
[INFO] |  +- com.google.code.gson:gson:jar:2.2.4:compile
[INFO] |  +- org.slf4j:slf4j-api:jar:1.7.5:compile
[INFO] |  +- org.apache.tephra:tephra-api:jar:0.15.0-incubating:compile
[INFO] |  +- org.apache.twill:twill-api:jar:0.14.0:compile
[INFO] |  |  +- org.apache.twill:twill-common:jar:0.14.0:compile
[INFO] |  |  \- org.apache.twill:twill-discovery-api:jar:0.14.0:compile
[INFO] |  \- javax.ws.rs:javax.ws.rs-api:jar:2.0:compile
[INFO] +- io.cdap.cdap:cdap-unit-test:jar:6.5.0-SNAPSHOT:test
[INFO] |  +- com.google.guava:guava:jar:13.0.1:provided
[INFO] |  +- io.cdap.cdap:cdap-test:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.cdap:cdap-proto:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  \- io.cdap.cdap:cdap-common:jar:6.5.0-SNAPSHOT:test
[INFO] |  +- io.cdap.cdap:cdap-app-fabric:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.cdap:cdap-data-fabric:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  |  +- io.cdap.cdap:cdap-metadata-spi:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  |  +- io.cdap.cdap:cdap-elastic:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  |  |  \- org.elasticsearch.client:elasticsearch-rest-high-level-client:jar:6.5.3:test
[INFO] |  |  |  |     +- org.elasticsearch:elasticsearch:jar:6.5.3:test
[INFO] |  |  |  |     |  +- org.elasticsearch:elasticsearch-core:jar:6.5.3:test
[INFO] |  |  |  |     |  +- org.elasticsearch:elasticsearch-secure-sm:jar:6.5.3:test
[INFO] |  |  |  |     |  +- org.elasticsearch:elasticsearch-x-content:jar:6.5.3:test
[INFO] |  |  |  |     |  |  +- org.yaml:snakeyaml:jar:1.17:test
[INFO] |  |  |  |     |  |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-smile:jar:2.8.11:test
[INFO] |  |  |  |     |  |  +- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.8.11:test
[INFO] |  |  |  |     |  |  \- com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:jar:2.8.8:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-core:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-analyzers-common:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-backward-codecs:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-grouping:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-highlighter:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-join:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-memory:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-misc:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-queries:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-queryparser:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-sandbox:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-spatial:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-spatial-extras:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-spatial3d:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.apache.lucene:lucene-suggest:jar:7.5.0:test
[INFO] |  |  |  |     |  +- org.elasticsearch:elasticsearch-cli:jar:6.5.3:test
[INFO] |  |  |  |     |  +- com.carrotsearch:hppc:jar:0.7.1:test
[INFO] |  |  |  |     |  +- com.tdunning:t-digest:jar:3.2:test
[INFO] |  |  |  |     |  +- org.hdrhistogram:HdrHistogram:jar:2.1.9:test
[INFO] |  |  |  |     |  +- org.apache.logging.log4j:log4j-api:jar:2.11.1:test
[INFO] |  |  |  |     |  \- org.elasticsearch:jna:jar:4.5.1:test
[INFO] |  |  |  |     +- org.elasticsearch.client:elasticsearch-rest-client:jar:6.5.3:test
[INFO] |  |  |  |     |  +- org.apache.httpcomponents:httpasyncclient:jar:4.1.2:test
[INFO] |  |  |  |     |  +- org.apache.httpcomponents:httpcore-nio:jar:4.4.5:test
[INFO] |  |  |  |     |  \- commons-logging:commons-logging:jar:1.1.3:test
[INFO] |  |  |  |     +- org.elasticsearch.plugin:parent-join-client:jar:6.5.3:test
[INFO] |  |  |  |     +- org.elasticsearch.plugin:aggs-matrix-stats-client:jar:6.5.3:test
[INFO] |  |  |  |     +- org.elasticsearch.plugin:rank-eval-client:jar:6.5.3:test
[INFO] |  |  |  |     \- org.elasticsearch.plugin:lang-mustache-client:jar:6.5.3:test
[INFO] |  |  |  |        \- com.github.spullara.mustache.java:compiler:jar:0.9.3:test
[INFO] |  |  |  +- org.apache.hadoop:hadoop-mapreduce-client-core:jar:2.6.0:provided
[INFO] |  |  |  \- org.apache.commons:commons-dbcp2:jar:2.6.0:test
[INFO] |  |  |     \- org.apache.commons:commons-pool2:jar:2.6.1:test
[INFO] |  |  +- io.cdap.cdap:cdap-system-app-api:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.cdap:cdap-master-spi:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.cdap:cdap-runtime-spi:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.cdap:cdap-security-spi:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.cdap:cdap-security:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- org.quartz-scheduler:quartz:jar:2.2.0:test
[INFO] |  |  +- org.quartz-scheduler:quartz-jobs:jar:2.2.0:test
[INFO] |  |  +- com.ning:async-http-client:jar:1.7.18:test
[INFO] |  |  +- org.apache.twill:twill-yarn:jar:0.14.0:test
[INFO] |  |  |  +- org.slf4j:jcl-over-slf4j:jar:1.7.5:test
[INFO] |  |  |  \- org.apache.hadoop:hadoop-hdfs:jar:2.6.0:test
[INFO] |  |  |     \- xerces:xercesImpl:jar:2.9.1:test
[INFO] |  |  |        \- xml-apis:xml-apis:jar:1.3.04:test
[INFO] |  |  +- io.cdap.cdap:cdap-client:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  |  +- commons-lang:commons-lang:jar:2.6:provided
[INFO] |  |  |  \- io.cdap.cdap:cdap-authentication-client:jar:1.4.0:test
[INFO] |  |  +- org.apache.hadoop:hadoop-mapreduce-client-common:jar:2.6.0:test
[INFO] |  |  |  +- org.apache.hadoop:hadoop-yarn-server-common:jar:2.6.0:test
[INFO] |  |  |  |  \- org.fusesource.leveldbjni:leveldbjni-all:jar:1.8:test
[INFO] |  |  |  +- com.google.protobuf:protobuf-java:jar:2.5.0:provided
[INFO] |  |  |  +- org.apache.hadoop:hadoop-annotations:jar:2.6.0:provided
[INFO] |  |  |  \- io.netty:netty:jar:3.6.2.Final:provided
[INFO] |  |  +- org.apache.hadoop:hadoop-yarn-api:jar:2.6.0:provided
[INFO] |  |  +- org.apache.hadoop:hadoop-yarn-client:jar:2.6.0:test
[INFO] |  |  |  +- commons-cli:commons-cli:jar:1.2:provided
[INFO] |  |  |  \- log4j:log4j:jar:1.2.17:provided
[INFO] |  |  +- org.apache.hadoop:hadoop-yarn-common:jar:2.6.0:provided
[INFO] |  |  |  +- commons-codec:commons-codec:jar:1.4:provided
[INFO] |  |  |  +- org.mortbay.jetty:jetty-util:jar:6.1.26:provided
[INFO] |  |  |  +- com.sun.jersey:jersey-client:jar:1.9:provided
[INFO] |  |  |  +- org.codehaus.jackson:jackson-core-asl:jar:1.9.13:provided
[INFO] |  |  |  +- org.codehaus.jackson:jackson-mapper-asl:jar:1.9.13:provided
[INFO] |  |  |  +- org.codehaus.jackson:jackson-jaxrs:jar:1.9.13:provided
[INFO] |  |  |  +- org.codehaus.jackson:jackson-xc:jar:1.9.13:provided
[INFO] |  |  |  \- commons-io:commons-io:jar:2.4:provided
[INFO] |  |  +- org.apache.hadoop:hadoop-mapreduce-client-jobclient:jar:2.6.0:test
[INFO] |  |  |  \- com.google.inject.extensions:guice-servlet:jar:3.0:test
[INFO] |  |  +- org.apache.hadoop:hadoop-mapreduce-client-app:jar:2.6.0:test
[INFO] |  |  +- org.apache.hbase:hbase-common:jar:1.0.1.1:test
[INFO] |  |  |  +- org.apache.hbase:hbase-annotations:jar:1.0.1.1:test
[INFO] |  |  |  |  \- jdk.tools:jdk.tools:jar:1.7:system
[INFO] |  |  |  +- commons-collections:commons-collections:jar:3.2.1:provided
[INFO] |  |  |  +- org.apache.htrace:htrace-core:jar:3.1.0-incubating:test
[INFO] |  |  |  \- com.github.stephenc.findbugs:findbugs-annotations:jar:1.3.9-1:test
[INFO] |  |  +- org.apache.commons:commons-compress:jar:1.18:provided
[INFO] |  |  +- com.jcraft:jsch:jar:0.1.54:provided
[INFO] |  |  +- org.apache.hive:hive-service:jar:1.2.1:test
[INFO] |  |  |  +- net.sf.jpam:jpam:jar:1.1:test
[INFO] |  |  |  +- org.apache.thrift:libfb303:jar:0.9.2:test
[INFO] |  |  |  +- org.apache.curator:curator-framework:jar:2.6.0:provided
[INFO] |  |  |  \- org.apache.curator:curator-recipes:jar:2.6.0:provided
[INFO] |  |  +- org.apache.avro:avro:jar:1.8.2:provided
[INFO] |  |  |  \- org.tukaani:xz:jar:1.5:provided
[INFO] |  |  \- org.slf4j:jul-to-slf4j:jar:1.7.5:test
[INFO] |  +- io.cdap.cdap:cdap-app-fabric:test-jar:tests:6.5.0-SNAPSHOT:test
[INFO] |  +- io.cdap.cdap:cdap-common:test-jar:tests:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.cdap:cdap-watchdog-api:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.cdap:cdap-formats:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.http:netty-http:jar:1.5.0:test
[INFO] |  |  |  \- commons-beanutils:commons-beanutils-core:jar:1.8.3:provided
[INFO] |  |  +- io.netty:netty-all:jar:4.1.16.Final:test
[INFO] |  |  +- com.google.inject:guice:jar:4.0:provided
[INFO] |  |  |  +- javax.inject:javax.inject:jar:1:provided
[INFO] |  |  |  \- aopalliance:aopalliance:jar:1.0:provided
[INFO] |  |  +- com.google.inject.extensions:guice-multibindings:jar:4.0:test
[INFO] |  |  +- com.google.inject.extensions:guice-assistedinject:jar:4.0:test
[INFO] |  |  +- org.ow2.asm:asm-commons:jar:7.1:test
[INFO] |  |  |  +- org.ow2.asm:asm:jar:7.1:test
[INFO] |  |  |  +- org.ow2.asm:asm-tree:jar:7.1:test
[INFO] |  |  |  \- org.ow2.asm:asm-analysis:jar:7.1:test
[INFO] |  |  +- org.apache.thrift:libthrift:jar:0.9.3:test
[INFO] |  |  +- org.apache.twill:twill-discovery-core:jar:0.14.0:test
[INFO] |  |  +- org.apache.twill:twill-core:jar:0.14.0:test
[INFO] |  |  |  +- io.netty:netty-buffer:jar:4.1.16.Final:test
[INFO] |  |  |  |  \- io.netty:netty-common:jar:4.1.16.Final:test
[INFO] |  |  |  +- io.netty:netty-codec-http:jar:4.1.16.Final:test
[INFO] |  |  |  |  \- io.netty:netty-codec:jar:4.1.16.Final:test
[INFO] |  |  |  +- io.netty:netty-handler:jar:4.1.16.Final:test
[INFO] |  |  |  |  \- io.netty:netty-transport:jar:4.1.16.Final:test
[INFO] |  |  |  |     \- io.netty:netty-resolver:jar:4.1.16.Final:test
[INFO] |  |  |  \- net.sf.jopt-simple:jopt-simple:jar:3.2:test
[INFO] |  |  +- org.apache.twill:twill-zookeeper:jar:0.14.0:test
[INFO] |  |  |  \- org.apache.zookeeper:zookeeper:jar:3.4.5:provided
[INFO] |  |  +- org.apache.hadoop:hadoop-common:jar:2.6.0:provided
[INFO] |  |  |  +- org.apache.commons:commons-math3:jar:3.1.1:provided
[INFO] |  |  |  +- xmlenc:xmlenc:jar:0.52:provided
[INFO] |  |  |  +- commons-httpclient:commons-httpclient:jar:3.1:provided
[INFO] |  |  |  +- commons-net:commons-net:jar:3.1:provided
[INFO] |  |  |  +- commons-el:commons-el:jar:1.0:provided
[INFO] |  |  |  +- net.java.dev.jets3t:jets3t:jar:0.9.0:provided
[INFO] |  |  |  |  \- com.jamesmurty.utils:java-xmlbuilder:jar:0.4:provided
[INFO] |  |  |  +- commons-configuration:commons-configuration:jar:1.6:provided
[INFO] |  |  |  |  \- commons-digester:commons-digester:jar:1.8:provided
[INFO] |  |  |  |     \- commons-beanutils:commons-beanutils:jar:1.7.0:provided
[INFO] |  |  |  +- org.apache.hadoop:hadoop-auth:jar:2.6.0:provided
[INFO] |  |  |  |  \- org.apache.directory.server:apacheds-kerberos-codec:jar:2.0.0-M15:provided
[INFO] |  |  |  |     +- org.apache.directory.server:apacheds-i18n:jar:2.0.0-M15:provided
[INFO] |  |  |  |     +- org.apache.directory.api:api-asn1-api:jar:1.0.0-M20:provided
[INFO] |  |  |  |     \- org.apache.directory.api:api-util:jar:1.0.0-M20:provided
[INFO] |  |  |  +- org.apache.curator:curator-client:jar:2.6.0:provided
[INFO] |  |  |  \- org.htrace:htrace-core:jar:3.0.4:provided
[INFO] |  |  +- org.bouncycastle:bcprov-jdk15on:jar:1.60:test
[INFO] |  |  +- org.bouncycastle:bcpkix-jdk15on:jar:1.60:test
[INFO] |  |  +- javax.servlet:javax.servlet-api:jar:3.0.1:test
[INFO] |  |  +- io.cdap.common:common-http:jar:0.13.0:test
[INFO] |  |  \- io.cdap.common:common-io:jar:0.13.0:test
[INFO] |  |     \- io.cdap.common:common-lang:jar:0.13.0:test
[INFO] |  +- io.cdap.cdap:cdap-watchdog:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  \- org.xerial.snappy:snappy-java:jar:1.1.1.7:provided
[INFO] |  +- io.cdap.cdap:cdap-explore:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- org.apache.hive:hive-jdbc:jar:1.2.1:test
[INFO] |  |  |  +- org.apache.hive:hive-common:jar:1.2.1:test
[INFO] |  |  |  |  +- joda-time:joda-time:jar:2.5:test
[INFO] |  |  |  |  +- log4j:apache-log4j-extras:jar:1.2.17:test
[INFO] |  |  |  |  +- org.apache.ant:ant:jar:1.9.1:test
[INFO] |  |  |  |  |  \- org.apache.ant:ant-launcher:jar:1.9.1:test
[INFO] |  |  |  |  \- org.json:json:jar:20090211:test
[INFO] |  |  |  +- org.apache.hive:hive-serde:jar:1.2.1:test
[INFO] |  |  |  |  \- net.sf.opencsv:opencsv:jar:2.3:test
[INFO] |  |  |  +- org.apache.hive:hive-shims:jar:1.2.1:test
[INFO] |  |  |  |  +- org.apache.hive.shims:hive-shims-common:jar:1.2.1:test
[INFO] |  |  |  |  +- org.apache.hive.shims:hive-shims-0.20S:jar:1.2.1:test
[INFO] |  |  |  |  +- org.apache.hive.shims:hive-shims-0.23:jar:1.2.1:test
[INFO] |  |  |  |  \- org.apache.hive.shims:hive-shims-scheduler:jar:1.2.1:test
[INFO] |  |  |  +- org.apache.httpcomponents:httpclient:jar:4.4:provided
[INFO] |  |  |  \- org.apache.httpcomponents:httpcore:jar:4.4:provided
[INFO] |  |  +- org.apache.hive:hive-metastore:jar:1.2.1:test
[INFO] |  |  |  +- com.jolbox:bonecp:jar:0.8.0.RELEASE:test
[INFO] |  |  |  +- org.apache.derby:derby:jar:10.10.2.0:test
[INFO] |  |  |  +- org.datanucleus:datanucleus-api-jdo:jar:3.2.6:test
[INFO] |  |  |  +- org.datanucleus:datanucleus-core:jar:3.2.10:test
[INFO] |  |  |  +- org.datanucleus:datanucleus-rdbms:jar:3.2.9:test
[INFO] |  |  |  +- commons-pool:commons-pool:jar:1.5.4:test
[INFO] |  |  |  +- commons-dbcp:commons-dbcp:jar:1.4:test
[INFO] |  |  |  +- javax.jdo:jdo-api:jar:3.0.1:test
[INFO] |  |  |  |  \- javax.transaction:jta:jar:1.1:test
[INFO] |  |  |  \- org.antlr:antlr-runtime:jar:3.4:test
[INFO] |  |  |     +- org.antlr:stringtemplate:jar:3.2.1:test
[INFO] |  |  |     \- antlr:antlr:jar:2.7.7:test
[INFO] |  |  \- org.apache.avro:avro-mapred:jar:1.8.2:test
[INFO] |  |     \- org.apache.avro:avro-ipc:jar:1.8.2:test
[INFO] |  |        +- org.mortbay.jetty:jetty:jar:6.1.22:test
[INFO] |  |        \- org.apache.velocity:velocity:jar:1.7:test
[INFO] |  +- io.cdap.cdap:cdap-explore-client:jar:6.5.0-SNAPSHOT:test
[INFO] |  +- io.cdap.cdap:cdap-explore-jdbc:jar:6.5.0-SNAPSHOT:test
[INFO] |  +- io.cdap.cdap:cdap-api-spark2_2.11:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  \- org.scala-lang:scala-library:jar:2.11.8:test
[INFO] |  +- io.cdap.cdap:cdap-spark-core2_2.11:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.cdap:cdap-spark-python:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- org.apache.spark:spark-core_2.11:jar:2.1.3:test
[INFO] |  |  |  +- org.apache.avro:avro-mapred:jar:hadoop2:1.7.7:test
[INFO] |  |  |  |  \- org.apache.avro:avro-ipc:jar:tests:1.7.7:test
[INFO] |  |  |  +- com.twitter:chill_2.11:jar:0.8.0:test
[INFO] |  |  |  |  \- com.esotericsoftware:kryo-shaded:jar:3.0.3:test
[INFO] |  |  |  |     +- com.esotericsoftware:minlog:jar:1.3.0:test
[INFO] |  |  |  |     \- org.objenesis:objenesis:jar:2.1:test
[INFO] |  |  |  +- com.twitter:chill-java:jar:0.8.0:test
[INFO] |  |  |  +- org.apache.xbean:xbean-asm5-shaded:jar:4.4:test
[INFO] |  |  |  +- org.apache.spark:spark-launcher_2.11:jar:2.1.3:test
[INFO] |  |  |  +- org.apache.spark:spark-network-common_2.11:jar:2.1.3:test
[INFO] |  |  |  |  \- com.fasterxml.jackson.core:jackson-annotations:jar:2.8.8:test
[INFO] |  |  |  +- org.apache.spark:spark-network-shuffle_2.11:jar:2.1.3:test
[INFO] |  |  |  +- org.apache.spark:spark-unsafe_2.11:jar:2.1.3:test
[INFO] |  |  |  +- org.apache.commons:commons-lang3:jar:3.5:test
[INFO] |  |  |  +- com.ning:compress-lzf:jar:1.0.3:test
[INFO] |  |  |  +- net.jpountz.lz4:lz4:jar:1.3.0:test
[INFO] |  |  |  +- org.roaringbitmap:RoaringBitmap:jar:0.5.11:test
[INFO] |  |  |  +- org.json4s:json4s-jackson_2.11:jar:3.2.11:test
[INFO] |  |  |  |  \- org.json4s:json4s-core_2.11:jar:3.2.11:test
[INFO] |  |  |  |     +- org.json4s:json4s-ast_2.11:jar:3.2.11:test
[INFO] |  |  |  |     +- com.thoughtworks.paranamer:paranamer:jar:2.6:test
[INFO] |  |  |  |     \- org.scala-lang:scalap:jar:2.11.0:test
[INFO] |  |  |  +- org.glassfish.jersey.core:jersey-client:jar:2.22.2:test
[INFO] |  |  |  |  +- org.glassfish.hk2:hk2-api:jar:2.4.0-b34:test
[INFO] |  |  |  |  |  +- org.glassfish.hk2:hk2-utils:jar:2.4.0-b34:test
[INFO] |  |  |  |  |  \- org.glassfish.hk2.external:aopalliance-repackaged:jar:2.4.0-b34:test
[INFO] |  |  |  |  +- org.glassfish.hk2.external:javax.inject:jar:2.4.0-b34:test
[INFO] |  |  |  |  \- org.glassfish.hk2:hk2-locator:jar:2.4.0-b34:test
[INFO] |  |  |  |     \- org.javassist:javassist:jar:3.18.1-GA:test
[INFO] |  |  |  +- org.glassfish.jersey.core:jersey-common:jar:2.22.2:test
[INFO] |  |  |  |  +- javax.annotation:javax.annotation-api:jar:1.2:test
[INFO] |  |  |  |  +- org.glassfish.jersey.bundles.repackaged:jersey-guava:jar:2.22.2:test
[INFO] |  |  |  |  \- org.glassfish.hk2:osgi-resource-locator:jar:1.0.1:test
[INFO] |  |  |  +- org.glassfish.jersey.core:jersey-server:jar:2.22.2:test
[INFO] |  |  |  |  +- org.glassfish.jersey.media:jersey-media-jaxb:jar:2.22.2:test
[INFO] |  |  |  |  \- javax.validation:validation-api:jar:1.1.0.Final:test
[INFO] |  |  |  +- org.glassfish.jersey.containers:jersey-container-servlet:jar:2.22.2:test
[INFO] |  |  |  +- org.glassfish.jersey.containers:jersey-container-servlet-core:jar:2.22.2:test
[INFO] |  |  |  +- com.clearspring.analytics:stream:jar:2.7.0:test
[INFO] |  |  |  +- io.dropwizard.metrics:metrics-jvm:jar:3.1.2:test
[INFO] |  |  |  +- io.dropwizard.metrics:metrics-json:jar:3.1.2:test
[INFO] |  |  |  +- io.dropwizard.metrics:metrics-graphite:jar:3.1.2:test
[INFO] |  |  |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.8.8:test
[INFO] |  |  |  |  \- com.fasterxml.jackson.core:jackson-core:jar:2.8.8:test
[INFO] |  |  |  +- com.fasterxml.jackson.module:jackson-module-scala_2.11:jar:2.8.8:test
[INFO] |  |  |  |  \- com.fasterxml.jackson.module:jackson-module-paranamer:jar:2.8.8:test
[INFO] |  |  |  +- org.apache.ivy:ivy:jar:2.4.0:test
[INFO] |  |  |  +- oro:oro:jar:2.0.8:test
[INFO] |  |  |  +- net.razorvine:pyrolite:jar:4.13:test
[INFO] |  |  |  +- net.sf.py4j:py4j:jar:0.10.7:test
[INFO] |  |  |  +- org.apache.spark:spark-tags_2.11:jar:2.1.3:test
[INFO] |  |  |  +- org.apache.commons:commons-crypto:jar:1.0.0:test
[INFO] |  |  |  \- org.spark-project.spark:unused:jar:1.0.0:test
[INFO] |  |  +- org.apache.spark:spark-streaming_2.11:jar:2.1.3:test
[INFO] |  |  +- org.apache.spark:spark-sql_2.11:jar:2.1.3:test
[INFO] |  |  |  +- com.univocity:univocity-parsers:jar:2.2.1:test
[INFO] |  |  |  +- org.apache.spark:spark-sketch_2.11:jar:2.1.3:test
[INFO] |  |  |  +- org.apache.spark:spark-catalyst_2.11:jar:2.1.3:test
[INFO] |  |  |  |  +- org.codehaus.janino:janino:jar:3.0.8:test
[INFO] |  |  |  |  +- org.codehaus.janino:commons-compiler:jar:3.0.8:test
[INFO] |  |  |  |  \- org.antlr:antlr4-runtime:jar:4.5.3:test
[INFO] |  |  |  +- org.apache.parquet:parquet-column:jar:1.8.1:test
[INFO] |  |  |  |  +- org.apache.parquet:parquet-common:jar:1.8.1:test
[INFO] |  |  |  |  \- org.apache.parquet:parquet-encoding:jar:1.8.1:test
[INFO] |  |  |  \- org.apache.parquet:parquet-hadoop:jar:1.8.1:test
[INFO] |  |  |     +- org.apache.parquet:parquet-format:jar:2.3.0-incubating:test
[INFO] |  |  |     \- org.apache.parquet:parquet-jackson:jar:1.8.1:test
[INFO] |  |  +- org.apache.spark:spark-mllib_2.11:jar:2.1.3:test
[INFO] |  |  |  +- org.apache.spark:spark-graphx_2.11:jar:2.1.3:test
[INFO] |  |  |  |  +- com.github.fommil.netlib:core:jar:1.1.2:test
[INFO] |  |  |  |  \- net.sourceforge.f2j:arpack_combined_all:jar:0.1:test
[INFO] |  |  |  +- org.apache.spark:spark-mllib-local_2.11:jar:2.1.3:test
[INFO] |  |  |  +- org.scalanlp:breeze_2.11:jar:0.12:test
[INFO] |  |  |  |  +- org.scalanlp:breeze-macros_2.11:jar:0.12:test
[INFO] |  |  |  |  +- com.github.rwl:jtransforms:jar:2.4.0:test
[INFO] |  |  |  |  +- org.spire-math:spire_2.11:jar:0.7.4:test
[INFO] |  |  |  |  |  \- org.spire-math:spire-macros_2.11:jar:0.7.4:test
[INFO] |  |  |  |  \- com.chuusai:shapeless_2.11:jar:2.0.0:test
[INFO] |  |  |  \- org.jpmml:pmml-model:jar:1.2.15:test
[INFO] |  |  |     \- org.jpmml:pmml-schema:jar:1.2.15:test
[INFO] |  |  \- org.apache.spark:spark-repl_2.11:jar:2.1.3:test
[INFO] |  |     +- org.scala-lang:scala-compiler:jar:2.11.8:test
[INFO] |  |     |  +- org.scala-lang.modules:scala-xml_2.11:jar:1.0.4:test
[INFO] |  |     |  \- org.scala-lang.modules:scala-parser-combinators_2.11:jar:1.0.4:test
[INFO] |  |     +- org.scala-lang:scala-reflect:jar:2.11.8:test
[INFO] |  |     \- jline:jline:jar:2.12:test
[INFO] |  +- io.cdap.cdap:cdap-tms:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.cdap:cdap-hbase-compat-base:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  |  \- io.cdap.cdap:cdap-hbase-spi:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- org.iq80.leveldb:leveldb:jar:uber:0.12:test
[INFO] |  |  |  \- org.iq80.leveldb:leveldb-api:jar:0.12:test
[INFO] |  |  \- io.cdap.cdap:cdap-storage-spi:jar:6.5.0-SNAPSHOT:test
[INFO] |  +- io.cdap.cdap:cdap-security:test-jar:tests:6.5.0-SNAPSHOT:test
[INFO] |  |  +- io.cdap.cdap:cdap-securestore-spi:jar:6.5.0-SNAPSHOT:test
[INFO] |  |  +- ch.qos.logback:logback-core:jar:1.0.9:test
[INFO] |  |  +- ch.qos.logback:logback-classic:jar:1.0.9:test
[INFO] |  |  +- org.eclipse.jetty:jetty-server:jar:8.1.15.v20140411:test
[INFO] |  |  |  +- org.eclipse.jetty.orbit:javax.servlet:jar:3.0.0.v201112011016:test
[INFO] |  |  |  +- org.eclipse.jetty:jetty-continuation:jar:8.1.15.v20140411:test
[INFO] |  |  |  \- org.eclipse.jetty:jetty-http:jar:8.1.15.v20140411:test
[INFO] |  |  |     \- org.eclipse.jetty:jetty-io:jar:8.1.15.v20140411:test
[INFO] |  |  +- org.eclipse.jetty:jetty-security:jar:8.1.15.v20140411:test
[INFO] |  |  +- org.eclipse.jetty:jetty-util:jar:8.1.15.v20140411:test
[INFO] |  |  +- org.eclipse.jetty:jetty-jaspi:jar:8.1.15.v20140411:test
[INFO] |  |  |  \- org.eclipse.jetty.orbit:javax.security.auth.message:jar:1.0.0.v201108011116:test
[INFO] |  |  +- org.eclipse.jetty:jetty-plus:jar:8.1.15.v20140411:test
[INFO] |  |  |  +- org.eclipse.jetty.orbit:javax.transaction:jar:1.1.1.v201105210645:test
[INFO] |  |  |  +- org.eclipse.jetty:jetty-webapp:jar:8.1.15.v20140411:test
[INFO] |  |  |  |  +- org.eclipse.jetty:jetty-xml:jar:8.1.15.v20140411:test
[INFO] |  |  |  |  \- org.eclipse.jetty:jetty-servlet:jar:8.1.15.v20140411:test
[INFO] |  |  |  \- org.eclipse.jetty:jetty-jndi:jar:8.1.15.v20140411:test
[INFO] |  |  |     \- org.eclipse.jetty.orbit:javax.mail.glassfish:jar:1.4.1.v201005082020:test
[INFO] |  |  |        \- org.eclipse.jetty.orbit:javax.activation:jar:1.1.0.v201105071233:test
[INFO] |  |  +- org.apache.geronimo.components:geronimo-jaspi:jar:2.0.0:test
[INFO] |  |  |  +- org.apache.geronimo.specs:geronimo-jaspic_1.0_spec:jar:1.1:test
[INFO] |  |  |  \- org.apache.xbean:xbean-reflect:jar:3.6:test
[INFO] |  |  +- org.jboss.resteasy:resteasy-servlet-initializer:jar:3.0.8.Final:test
[INFO] |  |  |  +- org.jboss.resteasy:resteasy-jaxrs:jar:3.0.8.Final:test
[INFO] |  |  |  |  +- org.jboss.spec.javax.annotation:jboss-annotations-api_1.1_spec:jar:1.0.1.Final:test
[INFO] |  |  |  |  +- javax.activation:activation:jar:1.1:test
[INFO] |  |  |  |  \- net.jcip:jcip-annotations:jar:1.0:test
[INFO] |  |  |  +- org.jboss.resteasy:async-http-servlet-3.0:jar:3.0.8.Final:test
[INFO] |  |  |  \- org.jboss.resteasy:jaxrs-api:jar:3.0.8.Final:test
[INFO] |  |  \- org.jboss.resteasy:resteasy-guice:jar:3.0.8.Final:test
[INFO] |  |     \- javax.annotation:jsr250-api:jar:1.0:test
[INFO] |  +- junit:junit:jar:4.11:test
[INFO] |  |  \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] |  +- org.apache.tephra:tephra-core:jar:0.15.0-incubating:test
[INFO] |  |  +- it.unimi.dsi:fastutil:jar:6.5.6:test
[INFO] |  |  \- io.dropwizard.metrics:metrics-core:jar:3.1.2:test
[INFO] |  \- com.sun.jersey:jersey-servlet:jar:1.18:test
[INFO] |     \- com.sun.jersey:jersey-server:jar:1.18:test
[INFO] |        \- com.sun.jersey:jersey-core:jar:1.18:test
[INFO] \- io.cdap.cdap:cdap-common-unit-test:jar:6.5.0-SNAPSHOT:test
```

Which only shows Guava in the `test` scope...